oi/CodeGen.cpp

```diff
@@ -983,6 +1069,10 @@ void CodeGen::addTypeHandlers(const TypeGraph& typeGraph, std::string& code) {
     } else if (const auto* con = dynamic_cast<const Container*>(&t)) {
       genContainerTypeHandler(config_.features, definedContainers_,
                               con->containerInfo_, con->templateParams, code);
+    } else if (const auto* cap = dynamic_cast<const CaptureKeys*>(&t)) {
+      genContainerTypeHandler(config_.features, definedContainers_,
+                              cap->containerInfo(),
+                              cap->container().templateParams, code);
```

Comment:
is this necessary? the topo sorter tests make it look like this will visit the container twice and get stopped by the set check

Reply:
The CaptureKeys node has a different ContainerInfo from the map that it contains. The ContainerInfo is copied and modified in the KeyCapture pass